File: MARCH/01-03-2026/Move All Zeroes to End.py
class Solution:
    def pushZerosToEnd(self, arr):
        last_free = 0
        for a in arr:
            if a:
                arr[last_free] = a
                last_free += 1
        for i in range(last_free, len(arr)):
            arr[i] = 0

File: 2026_GFG_POTD/MARCH/01-03-2026/Move All Zeroes to End.java
class Solution {   
    private static void swap(int[] arr, int a, int b) {
        int temp = arr[a];
        arr[a] = arr[b];
        arr[b] = temp;
    }
    
    void pushZerosToEnd(int[] arr) {
        int n = arr.length;
        int start = 0;
        for(int i=0;i<n;i++) {
            if(arr[i]!=0) {
                swap(arr, i, start++);
            }
        }
    }
}